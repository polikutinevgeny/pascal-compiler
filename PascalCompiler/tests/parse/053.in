type
   arry = array[0..200] of integer;
var
   a, b, i, n: integer;
   arr: arry;

begin
   repeat
      n := n - 1;;
      while n <> 1 do
         b := n shl 1;
      for i := 0 to n do
      begin
         arry[i] := arry[i] * b;
         arry[n - i] := 0;
      end;
   until a > 0;
end.