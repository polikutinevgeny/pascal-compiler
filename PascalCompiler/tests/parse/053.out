program 
  integer;
  real;
  char;
  arry[0..200]: array of integer;
  a: integer ;
  b: integer ;
  i: integer ;
  n: integer ;
  arr: arry ;
└─repeat
  ├─Greater
  │ ├─a
  │ └─0
  ├─Assignment statement
  │ ├─n
  │ └─Minus
  │   ├─n
  │   └─1
  ├─while
  │ ├─NEqual
  │ │ ├─n
  │ │ └─1
  │ └─Assignment statement
  │   ├─b
  │   └─Shl
  │     ├─n
  │     └─1
  └─For
    ├─i
    ├─0
    ├─n
    └─compound statement
      ├─Assignment statement
      │ ├─Index
      │ │ ├─arr
      │ │ └─i
      │ └─Asterisk
      │   ├─Index
      │   │ ├─arr
      │   │ └─i
      │   └─b
      └─Assignment statement
        ├─Index
        │ ├─arr
        │ └─Minus
        │   ├─n
        │   └─i
        └─0
